main/src/ca/uwaterloo/flix/language/phase/Stratifier.scala

```diff
@@ -728,6 +732,15 @@ object Stratifier {
     }
   }
 
+  /**
+    * Compare the labels by field, and used unification for type equality.
+    */
+  private def labelEquality(l1: Label, l2: Label)(implicit flix: Flix): Boolean =
```

Comment:
In the interest of performance, the following is better:

First check the predicate, denotation, and arity. If they match then:
Second check if the terms unify.